libs/ui/src/store/wallet/wallet.selectors.ts

```diff
@@ -119,27 +121,62 @@ export const useAccountAssetsSelector = () =>
     checkEquality
   );
 
-export const useGasTokenSelector = () => {
+export const useGasTokenSelector = (): TokenWithFiatBalance => {
   const publicKeyHash = useSelectedAccountPublicKeyHashSelector();
   const accountsGasTokens = useAccountsGasTokensSelector();
   const { gasTokenMetadata, chainId } = useSelectedNetworkSelector();
   const accountGasTokenSlug = getAccountTokensSlug(chainId, publicKeyHash);
+  const tokenMetadataSlug = getTokenMetadataSlug(chainId, GAS_TOKEN_ADDRESS);
+  const allTokensMarketInfo = useTokensMarketInfoSelector();
+
+  const fiatBalance = Number(
+    getDollarValue({
+      amount: isDefined(accountsGasTokens[accountGasTokenSlug]) ? accountsGasTokens[accountGasTokenSlug].data : '0',
+      price: isDefined(allTokensMarketInfo[tokenMetadataSlug]) ? allTokensMarketInfo[tokenMetadataSlug].price : 0,
+      decimals: gasTokenMetadata.decimals,
+      toFixed: false
+    })
+  );
 
   return useMemo(
     () => ({
       ...gasTokenMetadata,
       balance: accountsGasTokens[accountGasTokenSlug] ?? 0,
       tokenAddress: GAS_TOKEN_ADDRESS,
-      isVisible: true
+      isVisible: true,
+      fiatBalance
     }),
-    [chainId, accountsGasTokens, accountGasTokenSlug]
+    [chainId, accountsGasTokens, accountGasTokenSlug, allTokensMarketInfo]
```

Comment:
why do you have `allTokensMarketInfo` which is not used inside hook, seems like it should be replaced with `fiatBalance`